### ISSUE TITLE
[0.9.2] panic in tsdb decode by name

### ISSUE THREAD

**Issue by jhorwit2** [influxdb/influxdb#3623]
Randomly I got this panic. 
```
[http] 2015/08/11 08:21:46 10.15.5.228 - - [11/Aug/2015:08:21:46 -0400] POST /write?db=clientwatch&precision=ms&rp= HTTP/1.1 204 0 - Apache-HttpClient/4.4.1 (Java/1.8.0_45) 88ae486a-4023-11e5-a87a-000000000000 44.335675ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0x5b4344]

goroutine 142697 [running]:
github.com/influxdb/influxdb/tsdb.(*FieldCodec).DecodeByName(0x0, 0xc21ac533a0, 0x5, 0x7fb2821af74c, 0x26, 0x7fffffff, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/shard.go:1048 +0x74
github.com/influxdb/influxdb/tsdb.(*tagSetCursor).Next(0xc2091a17a0, 0x13f767dbdeb40000, 0x13f7b67070030000, 0xc20ac817a0, 0x1, 0x1, 0xc21ac53b70, 0x1, 0x1, 0x7fb2a037e8c0, ...)
	/root/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:581 +0x4fd
github.com/influxdb/influxdb/tsdb.func·004(0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:438 +0x183
github.com/influxdb/influxdb/tsdb.(*aggTagSetCursor).Next(0xc209c84370, 0xc20ac81858, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:496 +0x3a
github.com/influxdb/influxdb/influxql.MapCount(0x7fb298269ba0, 0xc209c84370, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/influxql/functions.go:247 +0x4e
github.com/influxdb/influxdb/tsdb.(*AggMapper).NextChunk(0xc220bb6000, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:448 +0x8a6
github.com/influxdb/influxdb/tsdb.(*StatefulMapper).NextChunk(0xc21f8a8ca0, 0xc20a74fc80, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/engine.go:54 +0x5c
github.com/influxdb/influxdb/tsdb.(*AggregateExecutor).execute(0xc20a74e2d0, 0xc21166e540)
	/root/go/src/github.com/influxdb/influxdb/tsdb/engine.go:413 +0x20e0
created by github.com/influxdb/influxdb/tsdb.(*AggregateExecutor).Execute
	/root/go/src/github.com/influxdb/influxdb/tsdb/engine.go:337 +0x5a

goroutine 1 [chan receive, 1056 minutes]:
main.(*Main).Run(0xc20808fa80, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:86 +0x73b
main.main()
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:38 +0xdc

goroutine 6 [syscall, 1057 minutes]:
os/signal.loop()
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 8 [IO wait, 1057 minutes]:
net.(*pollDesc).Wait(0xc208010450, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010450, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc2080103f0, 0x0, 0x7fb2a374ccc8, 0xc2080fe000)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20803ade0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20803ade0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080c1980, 0x7fb2a37531c8, 0xc20803ade0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tcp/mux.go:45 +0x57
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:298 +0x7a8

goroutine 17 [syscall, 1057 minutes, locked to thread]:
runtime.goexit()
	/root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1

goroutine 9 [chan receive, 1057 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20803ade8, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc2080c19b0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/meta/store.go:1855 +0x64
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc2080104d0)
	/root/go/src/github.com/hashicorp/raft/net_transport.go:346 +0x50
created by github.com/hashicorp/raft.NewNetworkTransport
	/root/go/src/github.com/hashicorp/raft/net_transport.go:138 +0x2c4

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc2080be1c0)
	/root/go/src/github.com/hashicorp/raft/raft.go:807 +0xc7c
github.com/hashicorp/raft.(*Raft).runLeader(0xc2080be1c0)
	/root/go/src/github.com/hashicorp/raft/raft.go:779 +0x74e
github.com/hashicorp/raft.(*Raft).run(0xc2080be1c0)
	/root/go/src/github.com/hashicorp/raft/raft.go:576 +0xc9
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)·fm()
	/root/go/src/github.com/hashicorp/raft/raft.go:241 +0x27
github.com/hashicorp/raft.func·011()
	/root/go/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/go/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 11 [select, 6 minutes]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc2080be1c0)
	/root/go/src/github.com/hashicorp/raft/raft.go:490 +0xf74
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runFSM)·fm()
	/root/go/src/github.com/hashicorp/raft/raft.go:242 +0x27
github.com/hashicorp/raft.func·011()
	/root/go/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/go/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc2080be1c0)
	/root/go/src/github.com/hashicorp/raft/raft.go:1594 +0x466
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runSnapshots)·fm()
	/root/go/src/github.com/hashicorp/raft/raft.go:243 +0x27
github.com/hashicorp/raft.func·011()
	/root/go/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/go/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 13 [chan receive, 1057 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20803adf0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc20800c280)
	/root/go/src/github.com/influxdb/influxdb/meta/store.go:470 +0x8f
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/root/go/src/github.com/influxdb/influxdb/meta/store.go:205 +0x1c2

goroutine 14 [select, 1057 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc2080aa0e0, 0xc208046540)
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:421 +0x14a
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:304 +0x919

goroutine 15 [select]:
github.com/influxdb/influxdb/tsdb.(*Shard).autoflusher(0xc2080300b0, 0xc208046120)
	/root/go/src/github.com/influxdb/influxdb/tsdb/shard.go:468 +0x39d
created by github.com/influxdb/influxdb/tsdb.func·010
	/root/go/src/github.com/influxdb/influxdb/tsdb/shard.go:149 +0x611

goroutine 16 [sleep]:
github.com/influxdb/influxdb/tsdb.(*Shard).Flush(0xc208030000, 0x77359400, 0x0, 0x0)
	/root/go/src/github.com/influxdb/influxdb/tsdb/shard.go:392 +0x2dc
github.com/influxdb/influxdb/tsdb.(*Shard).autoflusher(0xc208030000, 0xc20804
```

**Comment by jhorwit2** [comment]
It appears to be caused by this... possibly.

```
[write] 2015/08/11 09:25:37 write failed for shard 2167 on node 1: field type conflict: input field "userID" on measurement "dashboard" is type float64, already exists as type integer
```

**Comment by beckettsean** [comment]
Has the panic recurred? If you intentionally write mismatched type data does it occur?

**Comment by otoolep** [comment]
Believe fixed by #3646, please let us know if you see the issue with the next nightly build or 0.9.3.
